lib/core/services/log.service.ts

```diff
@@ -15,16 +15,27 @@
  * limitations under the License.
  */
 
+/* tslint:disable:no-console  */
+
 import { Injectable } from '@angular/core';
 import { AppConfigService } from '../app-config/app-config.service';
 import { logLevels, LogLevelsEnum } from '../models/log-levels.model';
+import { Observable } from 'rxjs/Observable';
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
 
 @Injectable()
 export class LogService {
 
     currentLogLevel: LogLevelsEnum = LogLevelsEnum.TRACE;
 
+    private onMessageSubject: BehaviorSubject<any>;
+    onMessage: Observable<any>;
```

Comment:
Note that we risk introducing memory leaks if we do not unsubscribe from such events. This service might prevent garbage collection for subscribers.